confluence-syntax-xhtml/src/main/java/org/xwiki/contrib/confluence/parser/xhtml/internal/wikimodel/LinkTagHandler.java

```diff
@@ -96,6 +99,17 @@ protected void end(TagContext context)
             link.setLabel(link.getAnchor());
         }
 
-        context.getScannerContext().onReference(link);
+        MacroTagHandler.ConfluenceMacro macro = (MacroTagHandler.ConfluenceMacro) context.getTagStack().getStackParameter(CONFLUENCE_CONTAINER);
+        if (macro != null && "include".equals(macro.name)) {
+            String ref = link.getDocument();
+            String space = link.getSpace();
+            if (space != null && !space.isEmpty()) {
+                ref = space + "." + ref;
+            }
+            context.getScannerContext().onMacro("display",
```

Comment:
Calling onMacro is really not LinkTagHandler's job IMO. Would be cleaner and much more generic to simply set the value of whatever is the currently parsed macro parameter (there might be several of those types of parameters in some macro and I don't see why we would limit the fix to the confluence macro with name "include") and let MacroTagHandler take care of the onMacro call.

Also, as I mentioned in https://jira.xwiki.org/browse/CONFLUENCE-75 (which seems to be exactly the same problem with a slightly different include macro name), the syntax parser should just have a better support of macro parameters of type link and the conversion to an XWiki display macro should be done in a MacroConverter.